pwndbg/wrappers/readelf.py

```diff
@@ -33,3 +34,49 @@ def _extract_jumps(line):
             return False
     except IndexError:
         return False
+
+@pwndbg.wrappers.OnlyWithCommand(cmd_name)
+def get_load_segment_info():
+    '''
+    Looks for LOAD sections by parsing the output of `readelf -l <binary>`
+    '''
+    local_path = pwndbg.file.get_file(pwndbg.proc.exe)
+    cmd = [get_jmpslots.cmd_path, "-l", local_path]
+    readelf_out = pwndbg.wrappers.call_cmd(cmd)
+
+    segments = []
+    load_found = False
+
+    # Output from readelf is 
+    # Type           Offset             VirtAddr           PhysAddr
+    #                FileSiz            MemSiz             Flags  Align
+    # LOAD           0x0000000000000000 0x0000000000000000 0x0000000000000000
+    #                0x0000000000000830 0x0000000000000830  R E    0x200000
+    # Account for this using two regular expressions
+    re_first = r"\s+LOAD\s+(0x[0-9A-Fa-f]+) (0x[0-9A-Fa-f]+) (0x[0-9A-Fa-f]+)"
+    re_secnd = r"\s+(0x[0-9A-Fa-f]+) (0x[0-9A-Fa-f]+)  (.)(.)(.)\s+(0x[0-9A-Fa-f]+)"
+    hex2int = lambda x: int(x, 16)
+
+    for l in readelf_out.splitlines():
+        if "LOAD" in l:
+            load_found = True
+            offset, vaddr, paddr = map(hex2int, re.match(re_first, l).groups())
+        elif load_found:
+            fsize, msize, read, write, execute, align = re.match(re_secnd, l).groups()
+            fsize, msize, align = map(hex2int, (fsize, msize, align))
+            read = True if read == "R" else False
+            write = True if write == "W" else False
+            execute = True if execute == "E" else False
+
+            segments.append({"Offset":   offset,
```

Comment:
Hm.. we might want to have a `class Segment`, like we do have `class Page`; however I am not sure if that is needed right now.

Reply:
Depends on your preference of who eventually will take care of readelf functionality. If you plan to incorporate pwntools on the long run, you will have to adopt to their code anyway ...

Reply:
We are going to merge with OR use pwntools on the long run :).